### ISSUE TITLE
Problem with multi-word custom resources

### ISSUE THREAD

**Issue by eric-michel** [TypeRocket/typerocket#151]
I have a custom resource called "Serial Number Ranges". However, I think TypeRocket has a bit of an issue dealing with custom resources that are made up of multiple words.

When I use tr_resource_pages, I originally did this:
`tr_resources_pages( 'SerialNumberRange', 'Serial Number Ranges' );`

But then the heading on, for instance, the Add view was "Add SerialNumberRange" without spaces.

Then I changed my code to:
`tr_resources_pages( 'Serial Number Range', 'Serial Number Ranges' );`

This made my headings show up the way I wanted.

I originally changed my model and controller files to be Serial_Number_Range.php and Serial_Number_RangeController.php respectively assuming TypeRocket would add underscores when looking for the resource. But I pretty quickly found out that TypeRocket doesn't underscore, it just eliminates spaces in this case (so it was looking for a controller called SerialNumberRangeController.php).

Long story medium, there's a problem with how TypeRocket looks at Models for stuff like this. If I name my model file "SerialNumberRange.php", the Add and Edit pages work just fine, but I get an error when trying to view the index:
`Fatal error: Uncaught Error: Call to a member function findAll() on null in...`

However, when I name my model file "Serial_Number_Range.php", the index pages works fine, but for Add or Edit I get:
`Fatal error: Method TypeRocket\Elements\Fields\Text::__toString() must not throw an exception, caught Error: Call to a member function getFillableFields() on null in...`

The only way I can make it all work is to have two model files and class names, one with underscores and one without. Ideally what I'd love to have is the tr_resources_pages function updated to have a $resource argument that is independent of $singular. Something like the following:
```
function tr_resource_pages($singular, $plural = null, array $settings = [], $resource = null)
{

    if ( ! $plural) {
        $plural = \TypeRocket\Utility\Inflect::pluralize($singular);
    }
    if( ! $resource) {
        $resource = $singular;
    }

    $menu_id = 'add_resource_' . \TypeRocket\Utility\Sanitize::underscore($singular);

    return tr_page($resource, 'index', $plural, $settings)->apply(
        tr_page($resource, 'edit', 'Edit ' . $singular)->useController()->addNewButton()->removeMenu(),
        tr_page($resource, 'show', $singular)->useController()->addNewButton()->removeMenu(),
        tr_page($resource, 'delete', 'Delete ' . $singular)->useController()->removeMenu(),
        tr_page($resource, 'add', 'Add ' . $singular)->useController()->setArgument('menu', 'Add New')->adminBar($menu_id, $singular, 'new-content' )
    )->addNewButton()->useController();
}
```
This way, I could use `tr_resource_pages( 'Serial Number Range', 'Serial Number Ranges', , 'SerialNumberRange' )` which would allow me to unify my model and controller filenames while having user-readable titles on the admin pages.

The above snippet should (hopefully) keep the function backwards compatible if you'd like to use it, but I'm a novice so I imagine you'd have a better implementation.

I reckon I'll just make my own custom version of this function for now, but I certainly hate to replicate functionality that's almost all the way there! I'd love to know if you decide to implement this in the future so I can move my code over.

Thanks!
EM

**Comment by kevindees** [comment]
Hey @eric-michel 

Thanks for the information on your issue. I'll see if I can reproduce the bug and find a good fix for it.

Thanks,
Kevin

**Comment by eric-michel** [comment]
@kevindees appreciate it! Absolutely loving TypeRocket so far. I'd be happy to contribute if you're looking for help extending functionality.

**Comment by brettvan** [comment]
Just want to add a "second" to this particular issue.  I would like to use multi-word custom resource names as well, but it does not seem possible at the moment.

I will try the suggested code change to the tr_resource_pages() function as I would really like the readability and consistency of naming in the admin interface.  Thanks for the good work on this too!

[UPDATE]
Suggested change works as expected, perhaps this is just a temporary fix to give the UI what we need.

**Comment by eric-michel** [comment]
@brettvan Glad it worked for you!

**Comment by kevindees** [comment]
Fixed in v4.0.3 using your approach.

https://github.com/TypeRocket/core/releases/tag/v4.0.3
